src/services/traineeService.ts

```diff
@@ -83,19 +83,38 @@ export const updateTraineeService = async (
     )
   }
 
-  if (
-    updates.coachId &&
-    currentCohort.coaches.find((coach) => coach.toString() === updates.coachId)
-  ) {
-    throw new CustomError(
-      COHORT_BAD_REQUEST,
-      "Coach is not part of the current cohort",
-      400,
+  if (updates.coachId) {
+    const isCoachInCohort = !currentCohort.coaches.some(
+      (coachId) => coachId.toString() === updates.coachId,
+    ) // this is negated for now since ther references of coach and trainees are incompatible for now
+
+    if (!isCoachInCohort) {
```

Comment:
I would suggest moving this implementation into a reusable function. such as `isInCohort`